src/main/java/io/redlink/more/data/configuration/GarminWellnessApiConfiguration.java

```diff
@@ -0,0 +1,24 @@
+package io.redlink.more.data.configuration;
+
+import io.redlink.more.data.garmin.wellness.ApiClient;
+import io.redlink.more.data.garmin.wellness.client.UserApiApi;
+import io.redlink.more.data.garmin.wellness.client.UserControllerApi;
+import io.redlink.more.data.model.garmin.UserAccessTokenWithData;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpHeaders;
+
+@Configuration
```

Comment:
Du verwendest `@Configuration` falsch!

Man lässt sich nicht die Configuration injection und ruft dann die Getter Methoden auf, sondern man macht eine `@Configuration` um dann die Getter mit `@Bean` annotieren. In den Klassen welche diese dann brauchen injected man nicht die `@Configuration` Klasse, sondern direkt den `@Bean`.

Wenn man beim Aufruf des getters immer eine neue Instanz des `@Bean` benötigt, dann kann man das über die Annotation des [Scope](https://docs.spring.io/spring-framework/reference/core/beans/factory-scopes.html) erreichen. Default ist `singleton` .. also alle bekommen die gleiche Instanz. Man kann das aber auch `prototype` oder für diese Config wohl relevant `request` oder `session` machen.